packages/rollup-full-node/src/app/web3-rpc-handler.ts

```diff
@@ -494,23 +496,57 @@ export class DefaultWeb3Handler
   }
 
   public async getLogs(ovmFilter: any): Promise<any[]> {
-    log.debug(`Requesting logs with filter [${JSON.stringify(ovmFilter)}].`)
     const filter = JSON.parse(JSON.stringify(ovmFilter))
+    // We cannot filter out execution manager events or else convertInternalLogsToOvmLogs will break.  So add EM address to address filter
     if (filter['address']) {
-      const codeContractAddress = await this.context.executionManager.getCodeContractAddress(
-        filter.address
-      )
-      filter['address'] = codeContractAddress
+      if (!Array.isArray(filter['address'])) {
+        filter['address'] = [filter['address']]
+      }
+      const codeContractAddresses = []
+      for (const address of filter['address']) {
+        codeContractAddresses.push(
+          await this.context.executionManager.getCodeContractAddress(address)
+        )
+      }
+      filter['address'] = [
+        ...codeContractAddresses,
+        this.context.executionManager.address,
+      ]
     }
+    // We cannot filter out execution manager events or else convertInternalLogsToOvmLogs will break.  So add EM topics to topics filter
+    if (filter['topics']) {
+      if (filter['topics'].length > 1) {
+        // todo make this proper error
+        const msg = `The provided filter ${filter} has multiple levels of topic filter.  Multi-level topic filters are currently unsupported by the OVM.`
```

Comment:
Should you log `${JSON.stringify(filter)}`? It's probably just going to print `[object Object]` otherwise.